src/chain.js

```diff
@@ -24,12 +24,12 @@ var map = require('./map');
  *
  *      var duplicate = n => [n, n];
  *      R.chain(duplicate, [1, 2, 3]); //=> [1, 1, 2, 2, 3, 3]
+ *
+ *      R.chain(R.append, R.head)([1, 2, 3]); //=> [1, 2, 3, 1]
  */
 module.exports = _curry2(_dispatchable('chain', _xchain, function chain(fn, monad) {
   if (typeof monad === 'function') {
-    return function() {
-      return monad.call(this, fn.apply(this, arguments)).apply(this, arguments);
-    };
+    return function(x) { return fn(monad(x))(x); };
```

Comment:
of all the constraints we try to satisfy, supporting `this` context is the first one i would throw overboard.